CDP4Composition/Services/RuleVerification/RuleVerificationService.cs

```diff
@@ -246,19 +251,35 @@ private void Execute(ISession session, UserRuleVerification userRuleVerification
                     break;
             }
 
-            this.UpdateExecutedOn(session, userRuleVerification);
-
             if (violations != null)
             {
-                userRuleVerification.Violation.AddRange(violations);
-
-                CDPMessageBus.Current.SendObjectChangeEvent(userRuleVerification, EventKind.Updated);
-
-                foreach (var ruleViolation in violations)
-                {
-                    CDPMessageBus.Current.SendObjectChangeEvent(ruleViolation, EventKind.Added);
-                }
+                userRuleVerification.Status = RuleVerificationStatusKind.FAILED;
+
+                IDisposable subscription = null;
+
+                //Listen for changes to the verification rule that will happen after UpdateExecutedOn in order to get the updated version.
+                //The violations must be added lastly as they are not persistent 
+                subscription = CDPMessageBus.Current.Listen<ObjectChangedEvent>(userRuleVerification)
+                                    .Where(objectChange => objectChange.EventKind == EventKind.Updated)                                    
+                                    .ObserveOn(RxApp.MainThreadScheduler)
+                                    .Subscribe( updated =>
+                                    {
+                                        //Only interested in a single update
+                                        subscription.Dispose();
+
+                                        var verification = updated.ChangedThing as UserRuleVerification;
+                                        verification.Violation.AddRange(violations);
+
+                                        CDPMessageBus.Current.SendObjectChangeEvent(verification, EventKind.Updated);
+
+                                        foreach (var ruleViolation in violations)
+                                        {
+                                            CDPMessageBus.Current.SendObjectChangeEvent(ruleViolation, EventKind.Added);
+                                        }
+                                    });
             }
+
```

Comment:
I think there should be an else {} implementation where userRuleVerification.Status is set to `RuleVerificationStatusKind.Passed`.

Reply:
Done